onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -33,6 +35,143 @@ CloudEndPointInvoker::CloudEndPointInvoker(const CloudEndPointConfig& config,
   }
 }
 
+// OpenAIInvoker
+class OpenAIInvoker : public CloudEndPointInvoker {
+ public:
+  OpenAIInvoker(const CloudEndPointConfig& config, const AllocatorPtr& allocator);
+  onnxruntime::Status Send(const CloudEndPointConfig& run_options,
+                           const InlinedVector<std::string>& input_names,
+                           gsl::span<const OrtValue> ort_inputs,
+                           const InlinedVector<std::string>& output_names,
+                           std::vector<OrtValue>& ort_outputs) const override;
+
+ private:
+  std::string uri_;
+  std::string model_name_;
+};
+
+OpenAIInvoker::OpenAIInvoker(const CloudEndPointConfig& config,
+                             const AllocatorPtr& allocator) : CloudEndPointInvoker(config, allocator) {
+  ReadConfig(kAzureUri, uri_);
+  ReadConfig(kAzureModelName, model_name_);
+}
+
+struct ResponseBuffer {
+  ResponseBuffer() = default;
+  ~ResponseBuffer() = default;
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(ResponseBuffer);
+
+  using Chunk = std::unique_ptr<char[]>;
+  std::list<Chunk> chunks_;
+
+  void Fill(std::string& s) const {
+    std::for_each(chunks_.begin(), chunks_.end(), [&](const Chunk& chunk) { s.append(chunk.get()); });
+  }
+};
+
+static size_t WriteResponseCallback(void* contents, size_t size, size_t nmemb, void* userp) {
+  size_t realsize = size * nmemb;
+  auto response = reinterpret_cast<struct ResponseBuffer*>(userp);
+  response->chunks_.push_back(std::make_unique<char[]>(realsize + 1));
```

Comment:
Can use std::string instead of char[].

Reply:
If we are sure to expect string to be returned, let's just go with a stringstream.